### PR TITLE
Update PowerShellGet to install the modules and scripts to proper locations in PowerShell Core on Windows 

### PR DESCRIPTION
**Pull request by bmanikm**
1.Update PowerShellGet to install the modules and scripts to proper locations in PowerShell Core on Windows #2021. 
2.Added few CI tests for PowerShellGet

### CODE REVIEW

**Comment by bmanikm** [comment]
@jianyunt Could you please review this?

File: src/Modules/Shared/PowerShellGet/PSModule.psm1 [hunk 1]
```diff
@@ -19,7 +19,7 @@ function IsLinux { $PSVariable = Get-Variable -Name IsLinux -ErrorAction Ignore;
 function IsOSX { $PSVariable = Get-Variable -Name IsOSX -ErrorAction Ignore; return ($PSVariable -and $PSVariable.Value) }
 function IsCoreCLR { $PSVariable = Get-Variable -Name IsCoreCLR -ErrorAction Ignore; return ($PSVariable -and $PSVariable.Value) }
 
-if(IsWindows)
+if((IsWindows) -and (-not (IsCoreCLR)))
```
Comment (jianyunt) [inline src/Modules/Shared/PowerShellGet/PSModule.psm1]: On Nano server, we will install modules to $PSHome now?

Comment (bmanikm) [inline src/Modules/Shared/PowerShellGet/PSModule.psm1]: @jianyunt  On Nano Server, IsCoreCLR returns $false, so this doesn't change the installation location on Nano Server.

**Comment by jianyunt** [comment]
:shipit:

File: src/Modules/Shared/PowerShellGet/PSModule.psm1 [hunk 2]
```diff
@@ -30,23 +30,30 @@ else
 
 if(IsWindows)
 {
-    try
-    {
-        $script:MyDocumentsFolderPath = [Environment]::GetFolderPath("MyDocuments")
-    }
-    catch
+    if(IsCoreCLR)
```
Comment (mirichmo) [inline src/Modules/Shared/PowerShellGet/PSModule.psm1]: I think this check isn't quite correct. This should be an "if inbox powershell" check. Inbox PowerShell for full and core CLRs should use the path on line 41. Non-inbox PowerShell should use the $HOME path on line 35.

Comment (bmanikm) [inline src/Modules/Shared/PowerShellGet/PSModule.psm1]: Thanks @mirichmo, I have incorporated your review feedback.

**Pull request merged**